test/unit/org/apache/cassandra/io/util/PathUtilsTest.java

```diff
@@ -0,0 +1,108 @@
+import org.junit.Test;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.serializers.MarshalException;
+
+import static org.apache.cassandra.config.CassandraRelevantProperties.CONSISTENT_DIRECTORY_LISTINGS;
+import static org.junit.Assert.*;
+
+public class PathUtilsTest
+{
+    private static File classTestDir;
+
+    @BeforeClass
+    public static void beforeClass()
+    {
+        classTestDir = FileUtils.getTempDir().resolve("PathUtilsTest");
+        PathUtils.createDirectoryIfNotExists(classTestDir.toPath());
+        classTestDir.deleteRecursiveOnExit();
+    }
+
+    @Test
+    public void testDeleteContent()
+    {
+        File testDir = classTestDir.resolve("testDeleteContent");
+        assertTrue(PathUtils.createDirectoryIfNotExists(testDir.toPath()));
+
+        File file1 = testDir.resolve("file1");
+        assertTrue(PathUtils.createFileIfNotExists(file1.toPath()));
+
+        File subdir = testDir.resolve("subdir");
+        assertTrue(PathUtils.createDirectoryIfNotExists(subdir.toPath()));
+
+        File subdir_file2 = subdir.resolve("file2");
+        assertTrue(PathUtils.createFileIfNotExists(subdir_file2.toPath()));
+
+        List<Path> testDirContents = PathUtils.listPaths(testDir.toPath());
+        assertEquals(2, testDirContents.size());
+        assertTrue(testDirContents.contains(file1.toPath()));
+        assertTrue(testDirContents.contains(subdir.toPath()));
+
+        PathUtils.deleteContent(testDir.toPath());
+        assertTrue(testDir.exists());
+        assertTrue(PathUtils.listPaths(testDir.toPath()).isEmpty());
+    }
+
+    @Test
+    public void testListPaths()
+    {
+        File testDir = classTestDir.resolve("testListPaths");
+        assertTrue(PathUtils.createDirectoryIfNotExists(testDir.toPath()));
+        File file1 = testDir.resolve("file1");
+        assertTrue(PathUtils.createFileIfNotExists(file1.toPath()));
+
+        List<Path> testDirContents = PathUtils.listPaths(testDir.toPath());
+        assertNotNull(testDirContents);
+        assertTrue(testDirContents.size() >= 1);
+        assertTrue(testDirContents.contains(file1.toPath()));
+    }
+
+    @Test
+    public void testListPaths_NoSuchFile()
+    {
+        File testDir = classTestDir.resolve("testListPaths_NoSuchFile");
+        File doesNotExist = testDir.resolve("doesNotExist");
+        assertFalse(doesNotExist.exists());
+        assertTrue(PathUtils.listPaths(doesNotExist.toPath()).isEmpty());
+    }
+
+    @Test
+    public void testListPaths_NotDirectory()
+    {
+        File testDir = classTestDir.resolve("testListPaths_NotDirectory");
+        assertTrue(PathUtils.createDirectoryIfNotExists(testDir.toPath()));
+        File file1 = testDir.resolve("file1");
+        assertFalse(file1.exists());
+        assertTrue(PathUtils.createFileIfNotExists(file1.toPath()));
+        assertTrue(file1.exists());
+        assertTrue(PathUtils.listPaths(file1.toPath()).isEmpty());
+    }
+}
```

Comment:
I don't know if it is expected to always have a new line at the end of file or not. Here it is missing.

Reply:
My understanding is a new line is expected, so I will add it